r/tests/testthat/test-parquet.R

```diff
@@ -121,6 +121,21 @@ test_that("Factors are preserved when writing/reading from Parquet", {
   expect_identical(df, df_read)
 })
 
+test_that("Lists are preserved when writing/reading from Parquet", {
+  bool <- list(logical(0), NA, c(TRUE, FALSE))
+  int <- list(integer(0), NA_integer_, 1:4)
+  num <- list(numeric(0), NA_real_, c(1, 2))
+  char <- list(character(0), NA_character_, c("itsy", "bitsy"))
+  df <- tibble::tibble(bool = bool, int = int, num = num, char = char)
+
+  pq_tmp_file <- tempfile()
+  on.exit(unlink(pq_tmp_file))
+
+  write_parquet(df, pq_tmp_file)
+  df_read <- read_parquet(pq_tmp_file)
+  expect_identical(df, df_read)
```

Comment:
You're hitting a misfeature in `dplyr` so let's work around it. See https://github.com/apache/arrow/blob/master/r/tests/testthat/helper-expectation.R#L26-L37

```suggestion
  expect_equivalent(df, df_read)
```